core/basic-authorship/src/basic_authorship.rs

```diff
@@ -29,12 +29,9 @@ use client::{
 use codec::Decode;
 use consensus_common::{self, evaluation};
 use primitives::{H256, Blake2Hasher, ExecutionContext};
-use runtime_primitives::traits::{
-	Block as BlockT, Hash as HashT, Header as HeaderT, ProvideRuntimeApi,
-	DigestFor,
-};
-use runtime_primitives::generic::BlockId;
-use runtime_primitives::ApplyError;
+use sr_primitives::traits::{ Block as BlockT, Hash as HashT, Header as HeaderT, ProvideRuntimeApi, DigestFor};
```

Comment:
Space and all 3 lines can be merged.

Reply:
Well, actually do we have a rule for this? I am fine with merging such things if it is super short but at some point, it turns into multi-line imports and it is like:
```
use CRATE::{
   mod1, mod, mod, mod, mod, mod, mod, mod, sub_mod::{mod, mod},  mod, mod, mod,
    mod, mod, mod, sub_sub_mod::{foo},  mod, mod, mod, mod,
}
```

which makes it really hard to _easily see_ what is imported what is being _imported from which module within the crate_. Hence, if you agree, I would put the guideline as:
- One newline per submodule import, except if it is obviously short and too few.

Reply:
we can do also:
```
use CRATE::{
    sub_mod::{
          type, type, type, type, type, type, type, type, type
    }
    type, type, type, type, type, type, type, type, type

Reply:
I am personally in favor of a new import statement:

```
use CRATE:{type, type, type, type, type, type, type, type, type};
use CRATE::sub_mod::{type, type, type, type, type, type, type, type, type};
```

Since I think imports are not like typical code where you break to the next line and indent etc. (just looks awkward to me when you indent an import statement xD) and what I propose is less like to need breaking the line. 

But I am in favor of generally having a guideline for it; no strong opinion on a style.

Reply:
yes maybe @bkchr unrequest changes until we commit to some guideline on this ?